$MirOS: ports/devel/cvs/patches/patch-src_tag_c,v 1.3 2010/09/15 20:57:03 tg Exp $
--- src/tag.c.orig	Sun Sep  4 00:27:22 2005
+++ src/tag.c	Thu Sep 16 00:10:47 2010
@@ -20,6 +20,8 @@
 #include "cvs.h"
 #include "save-cwd.h"
 
+__RCSID("$MirOS: ports/devel/cvs/patches/patch-src_tag_c,v 1.3 2010/09/15 20:57:03 tg Exp $");
+
 static int rtag_proc (int argc, char **argv, char *xwhere,
 		      char *mwhere, char *mfile, int shorten,
 		      int local_specified, char *mname, char *msg);
@@ -310,6 +312,7 @@ posttag_proc (const char *repository, co
      * %b = branch mode = "?" (delete ops - unknown) | "T" (branch)
      *                    | "N" (not branch)
      * %c = cvs_cmd_name
+     * %I = commit ID
      * %p = path from $CVSROOT
      * %r = path from root
      * %{sVv} = attribute list = file name, old version tag will be deleted
@@ -333,6 +336,7 @@ posttag_proc (const char *repository, co
 			      "b", "c", delete_flag
 			      ? '?' : branch_mode ? 'T' : 'N',
 			      "c", "s", cvs_cmd_name,
+			      "I", "s", global_session_id,
 #ifdef SERVER_SUPPORT
 			      "R", "s", referrer ? referrer->original : "NONE",
 #endif /* SERVER_SUPPORT */
@@ -747,6 +751,7 @@ pretag_proc (const char *repository, con
      * %b = branch mode = "?" (delete ops - unknown) | "T" (branch)
      *                    | "N" (not branch)
      * %c = cvs_cmd_name
+     * %I = commit ID
      * %p = path from $CVSROOT
      * %r = path from root
      * %{sVv} = attribute list = file name, old version tag will be deleted
@@ -770,6 +775,7 @@ pretag_proc (const char *repository, con
 			      "b", "c", delete_flag
 			      ? '?' : branch_mode ? 'T' : 'N',
 			      "c", "s", cvs_cmd_name,
+			      "I", "s", global_session_id,
 #ifdef SERVER_SUPPORT
 			      "R", "s", referrer ? referrer->original : "NONE",
 #endif /* SERVER_SUPPORT */
@@ -1235,14 +1241,14 @@ tag_fileproc (void *callerdat, struct fi
     else if (strcmp (version, "0") == 0)
     {
 	if (!quiet)
-	    error (0, 0, "couldn't tag added but un-commited file `%s'",
+	    error (0, 0, "couldn't tag added but un-committed file `%s'",
 	           finfo->file);
 	goto free_vars_and_return;
     }
     else if (version[0] == '-')
     {
 	if (!quiet)
-	    error (0, 0, "skipping removed but un-commited file `%s'",
+	    error (0, 0, "skipping removed but un-committed file `%s'",
 		   finfo->file);
 	goto free_vars_and_return;
     }
@@ -1450,6 +1456,7 @@ val_fileproc (void *callerdat, struct fi
  *   		2. If IDB is non-NULL and val-tags cannot be opened for write.
  *   		   This allows callers to ignore the harmless inability to
  *   		   update the val-tags cache.
+ *		3. If CVSREADONLYFS is set (same as #2 above).
  *   false	If the file could be opened and the tag is not present.
  */
 static int is_in_val_tags (DBM **idb, const char *name)
@@ -1458,6 +1465,10 @@ static int is_in_val_tags (DBM **idb, co
     char *valtags_filename;
     datum mytag;
     int status;
+
+    /* do nothing if we know we fail anyway */
+    if (readonlyfs)
+      return 1;
 
     /* Casting out const should be safe here - input datums are not
      * written to by the myndbm functions.
